test/app/model/ChessEngine.spec.ts

```diff
@@ -445,3 +444,187 @@ describe('Return player moves without those that cause his king to check', () =>
     );
   });
 });
+
+describe('Castling can be done only if neither king nor rook has moved and none of square which the king is passing by is under attack and no piece between the king and the rook', () => {
+  const whiteKing = new King(Side.WHITE);
+  const whiteRookA1 = new Rook(Side.WHITE);
+  const whiteRookH1 = new Rook(Side.WHITE);
```

Comment:
Nie wiem czy jeśli dorabiamy nazwę np "white Rook A1" to nie powinniśmy w tym samym bloku wrzucić na planszę.
I czy nie lepiej jednak usunąć to "A1" - @nowakprojects ?

Ale podoba mnie się twoja kreatywność. Fajnie i pomysłowo tego użyłaś

Reply:
A no racja, troche to dziwne :) 
W ogole sporo uzywamy w testach. Moze warto zrobic stale do wszystkich testow modelu dostepne whiteRook / blackRook / whiteKnight / blackKnight itd :) ?

Reply:
Tego chyba już lepiej nie zmieniać we wszystkich testach. Zostawmy to na kiedyś indziej. Tamte są już ok, a to by się przydał osobny PR

Reply:
Prawda to, Piotrze "Strażniku Małych PRów" :D

Reply:
Sądziłam, że nie bardzo mogę użyć jednej wieży w dwóch miejscach na szachownicy, stąd A1/H1 - w niektórych case'ach sprawdzam obydwa przypadki.